### DISCUSSION TITLE
XML Schema support?

### DISCUSSION THREAD

**Discussion by Cicorione** [helix-editor/helix#9100] [Q&A]
Hi there,

I am interested in learning to use Helix and I wonder if it supports XML Schema. I am not a coder I am trying to learn a software based on LuaTeX and XML Schema is a requirement. I'd like to learn a modern editor that works on a TUI interface with a modern approach.

Thanks... 🙏🙏🙏

**Comment by Cicorione** [comment]
The official page says Helix support LSP:

https://langserver.org/

But it looks like XML is not supported yet...

https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers

☹️

**Comment by Rudxain** [comment]
Solution: [LemMinX](/eclipse/lemminx) (see #4518). Unfortunately, it depends on JDK, which is very bloated compared to NodeJS or CPython

**Reply by Cicorione** [reply]
I see... 😔

**Reply by Rudxain** [reply]
Wait, there's hope! [look at what RH did](https://developers.redhat.com/blog/2021/03/12/no-more-java-in-vscode-xml-0-15-0)

**Reply by Cicorione** [reply]
I tend to not use `vscode-xml` because it fills my ram very quickly, perhaps because it finds JDK on my systems and therefore it uses java instead.

As an alternative option I use: `Scholarly XML` which doesn't requires java: https://github.com/raffazizzi/vscode-sxml

But `Scholarly XML` does only `schema` validation, therefore you want to couple with other plugins that helps managing XML, on the contrary `vscode-xml` has additional features such as auto-tag-closing, beautify, etc...

Either way what I didn't understand are those "LSP modules" agnostic? Can I use them on Helix, or must the "LSP Module" be built explicitly for Helix?

Thanks! 🙏

**Reply by Rudxain** [reply]
I couldn't find anything about LSP "modules". But, if something _claims_ to be an LSP, it _should_ be compatible with any editor that supports the protocol.

I found this part ambiguous:
> Eclipse LemMinX is the language server that provides vscode-xml's XML editing features. By **creating a language server, we can provide XML editing capabilities** not only to Visual Studio Code (VS Code) but also to other text editors, such as Sublime, Eclipse IDE, Emacs, and Vim.

It kinda seems like RH created their own LSP, but RH could also be talking from a generic POV that fits Eclipse's POV. So I'm unsure if they refer to the existing LSP that depends on Java or a standalone LSP that they created for the extension

**Reply by Cicorione** [reply]
Sorry I don't know how to call them, those are VSCode extension but also LSP...
And I don't know if those extensions are compatible with other programs...  🤷‍♂️

**Reply by Rudxain** [reply]
oh ok :(

**Comment by rnwst** [comment]
If you want to get an XML language server working in Helix, install [`lemminx`](https://github.com/eclipse-lemminx/lemminx) and add the following to your `languages.toml`:
```toml
[language-server.lemminx]
command = "lemminx"

[[language]]
name = "xml"
language-servers = ["lemminx"]
auto-format = true
```

**Comment by ChristopherRabotin** [comment]
The simplest approach is to set the open file language to HTML, as per https://github.com/helix-editor/helix/issues/3479#issuecomment-1804191224.

`:set-language html`
